source/shared/cpp/ObjectModel/ChoicesData.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "ChoicesData.h"
+#include "ParseContext.h"
+#include "ParseUtil.h"
+
+using namespace AdaptiveCards;
+
+ChoicesData::ChoicesData()
+{
+}
+
+std::string ChoicesData::Serialize() const
+{
+    return ParseUtil::JsonToString(SerializeToJsonValue());
+}
+
+Json::Value ChoicesData::SerializeToJsonValue() const
+{
+    Json::Value root;
+
+    if (!m_type.empty())
+    {
+        root[AdaptiveCardSchemaKeyToString(AdaptiveCardSchemaKey::Type)] = GetType();
```

Comment:
please consider renaming GetType() as it confuses with GetType() of AdaptiveElement.